src/battle_ai_main.c

```diff
@@ -2693,6 +2694,11 @@ static s32 AI_DoubleBattle(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
                 ADJUST_SCORE(-10);
             break;
         }
+
+        // Don't use sleep effect if partner will activate sleep clause
+        if (IsEffectBlockedBySleepClause(partnerEffect) && IsEffectBlockedBySleepClause(effect))
+            ADJUST_SCORE(-20);
```

Comment:
Actually, why is this handled like that instead of the same way, let's say TrickRoom is handled? You could just use the `EFFECT_SLEEP` and `EFFECT_YAWN` cases used in this function and check for partner.

Reply:
The thought was that this is less invasive in a function that doesn't otherwise care about sleep clause and means that if anything else needs to reference sleep clause triggering effects for whatever reason in the future, there's on centralized place to add those effects.

I'm happy to add B_FLAG_SLEEP_CLAUSE and partner checks under explicit EFFECT_SLEEP and EFFECT_YAWN it just seemed more invasive to me whereas this felt cleaner and more futureproofed. I think it's subjective and I am happy to defer to your judgement if you prefer :)

Reply:
I don't think it is really that invasive and think it adds clarity instead. When I search for `EFFECT_SLEEP` / `EFFECT_YAWN` it would be better to have all checks in one place instead of scattered around.

Reply:
Addressed, thanks for clarifying :)